# MSC-Project
Whole genome analysis of RVA G1P[8] strain before and after vaccine introduction in Kenya
## Introduction
G1P[8] is among the most predominant and medically important RVA strain in human infections, globally. This strain accounts for approximately 29% of all the circulating RVA strains in Africa and was responsible for ~47% of all diarrhea infections in a multicenter study in Kenya[11]. The implementation of the monovalent Rotarix vaccine in NIP in Kenya resulted to a significant reduction in hospitalization cases caused by RVA strains. Interestingly, in the first year of vaccine introduction, there was continued dominance of genotype G1P[8] whose pre-valence declined in the subsequent years. However, increase in prevalence of non-G1P[8] strains (G2P[4], G3P[8], G3P[6] and G9P[8])after vaccine introduction have been reported as well. Such emergencies and increase in prevalence of RVA strains after vaccine introduction emphasize the need to monitor the impact of RVA vaccines on these circulating strains in the population. For this to be achieved, the availability of sufficient complete genome data before and after vaccine introduction will inform on the effectiveness of the vaccine against the diverse and evolutionary complex RVA strains.  Here, we employed high-throughput primer-independent direct genome and primer-dependent genome sequencing methods from stool samples to infer epidemiology, characterize and determine RVA diversity of G1P[8] RVA strains in rural Kenya, Kilifi, before and after vaccine introduction.

## Objectives

## Approach/Questions
Whats constellations

How is the local diversity/each segment\n
Global Diversity
CLustering patterns
Vaccinated vs unvaccinated
Pre and post-vaccine
Cytotoxic T-cells epitopes and B cell epitops
Selection pressure
Amino acid changes
Time Trees
Write-Up 
PHD Applications

## Methods
### Selection Pressure


