
# MSC-Project
Whole genome analysis of RVA G1P[8] strain before and after vaccine introduction in Kenya
## Abstract

Group A rotaviruses (RVA) are a leading cause of severe acute gastroenteritis in young children worldwide.  RVA clinical isolates show extensive genetic variability especially within the two outer capsid proteins named, VP7 (G) and VP4 (P) that are key targets of human immune responses. These two proteins form the basis for its binary (GP) RVA classification system. Kenya introduced the monovalent Rotarix® vaccine (derived from a live attenuated G1P[8] rotavirus strain) in July 2014.  This vaccine is expected to provide homotypic and heterotypic protection against circulating RVA strains. However, RVA vaccines including Rotarix® have shown lower efficacy in developing countries compared to developed countries for reasons that remain unclear to date. Continual RVA surveillance in Kilifi (coastal Kenya), since 2010, has shown a high prevalence of G1P[8] strains in both pre- and post-vaccination periods. Analysis of whole genome sequence data has potential to provide insight on the reasons underlying the vaccine underperformance and local persistence of G1P[8] strains despite ongoing nation-wide Rotarix® vaccination.  Genomic mutations, frequent genetic reassortment, and increased vaccine mediated-immunity might lead to antigenic evolution and development of vaccine escape strains. Further, close monitoring of G1P[8] is needed to provide insight into the origin of continued RVA cases as incidence declines. During this proposed project, I will undertake a whole-genomic analysis of G1P[8] strains collected in Kilifi Kenya before and after the Rotarix® vaccination programme took effect to determine the patterns of genome-wide variation of G1P[8] strains and temporal patterns of RVA phylogeny. This will advance the understanding of the impact of vaccination on RVA genetic diversity and transmission, particularly G1P[8] genotype.

## Problem Statement
RVA has been associated with high morbidity and mortality among children in developing countries. It is estimated that two thirds of RVA associated deaths occur in sub-Saharan Africa. Contrary to results from the developed countries, developing countries have experienced only modest vaccine performance and so far, the underlying reason for this underperformance remain uncertain. For long, this burden has been associated predominantly by RVA G1P[8] strains.  While a vaccine against RVA is available, it is not clear whether the vaccine protects equally against all lineages within genotype G1P[8] and how this affects its performance. Additionally, due to their segmented nature, RVAs can undergo reassortments in addition to point mutations that can result in amino acid changes, which may lead to increased antigenic variation conferring immune escape by some G1P[8] strains and eventually lower vaccine performance. However, despite, the significance of circulating G1P[8] strains, there is limited understanding on the evolutionary dynamics and impact of the vaccine on the genetic variability of this strain. 


## Main Objectives

To investigate the genomic characteristics of RVA G1P[8] strains circulating in rural Kenya and investigate their evolutionary dynamics in pre- and post-vaccine periods.

## Specific Objective

•	To investigate the complete genome constellations and genetic diversity of RVA G1[8] strains circulating before and after vaccine introduction in Kilifi.

•	To determine the phylogenetic and evolutionary relationship between the coastal Kenya G1P[8] strains and contemporaneous G1P[8] strains collected around the globe.

•	To determine the origins, and spread patterns of G1P[8] strains in Kilifi,  rural coastal Kenya in the pre- and post-rotavirus vaccine introduction era.


## Methods
Genome assembly  
Multiple seqeunce alignments  
Phylogenetic Inferences

## Results

#### G1P[8] prevalence in Kilifi

<img src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/Genotype_prevalence.png" align="center" width="750" height="500">
Table 1 <br>
<br>
Table 1 above shows the observed genogroup constellations of the G1P[8] strains.
While there was variation in prevalence of G1P[8] strains, they were continoiusly observed throughout the surveillance period. It was the dominant genotype in 2011, 2013, 2015, and 2017. It caused up to 47% of all RVA infections in Kenya.

### Objective 1: 

#### What are the genomic constellations of the observed G1P[8]

All possed a Wa-like genotype constellation, except two post-vaccine strains that; one possed a DS-1 like genotype E2 on the NSP4 gene segment, and the other possed a T6 genotype on the NSP3 gene segment. Potential rearssortants.

<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/Constellations.png" align="center">

### Objective 2:

### a. How is the local diversity/each segment/clustering patterns
<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/Local_diversity_claster_patterm.png" align="center">

Distinct clustering of pre and post vaccine strains in all the segments. <br>
All Gene segmenets partitioned into multiple clusters. Pre-vaccine strains formed multiple clusters in genes VP1, VP3, NSP1, NSP2 NSP3. <br>
Post-vaccines formed single clusters across all the segments except for segments NSP3 and NSP4 (different genogroup types). <br>
A single post-vaccine strain from 2015 clustered with pre-vaccine strains across all the 11 segments. Indicative of countinued circulation of pre-vaccine after vaccine introduction.<br>

**Violin plot representing pairwise distance within local strains**

<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/Genetic_diversity_local.png" align="center" width='650' height='650'>
  
Shows pairwise genetic distances within each gene, from Kilifi data <br>
Genetic distances supports corcirculation of multiple clusters of the G1P[8] strains

#### b. Whats the genetic relationship between local strains and Global contemporaneous strains plus the vaccine strain
<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/VP7_VP4_VP5_VP1.png" align="center" width='650' height='550'>
<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/VP2_VP3_NSP1_NSP2.png" align="center" width='650' height='550'>
<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/NSP3_NSP4_NSP5_2.png" align="center" width='750' height='600'>
    
Local strains cluster with similar genotypes observed across the world. <br>
The vaccine strain clustered distantyl from local strains <br>

**Indicate lineages for the G and P trees**

#### T6 Genotype

<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/T6.aln.treefile.png" align="center">
    
 Phylogeny of genotype T6 <br>
 Closest to a T6 genotype isolated from goat in Uganda 2014 and bovine from Italy . <br> 
 Potential rearsortant strain or zoonotic transmission
 
    
#### E2 Genotype

<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/E2.aln.nwk.png" align="center">
    
 Clustred with rearsortant strains from Vietnam and Japan <br>
 Potential rearsortant between DS-1-like and Wa-like strains.
 
### c. Whats the 'immunogenic'significance of the observed mutations?
#### i. Selection Pressure
Refers to determining evolutionary rate of protein coding genes - which is based on calculation the ratio of dN/dS substitutions. 
dN/dS < 1 - purifying selection, dN/dS =1 neutral selection, dN/dS > 1 - positive selection. The ratio summarizes evolutionary rate of genes and can be an informative feature in identifying genes which are the most conserved and those that are less conserved(undergone adaptive evolution). The rate of change indicates selective pressure occuring at non-synonymous sites.

 **Steps**

1. Analysis carried out in Datamonkey <br>  
2. Use multiple models in the website (FEL, FUBAR, mixed effects model of episodic selection (MEME)) <br>
3. Sequences submitted to Datamonkey for predictions which are hereby recorded for all the segment
      <image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/Sites_selection_pressure.png" align="center">
  
None of the pre and post vaccine changes were detected to be under any selection
  
<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/alignment.png" align="center" width='800' height='550'>




    
#### ii. Comparison of B-cell epitopes between the vaccine strain and the locally observed strains <br>
Amino acid identity between the epitopes and the vaccine implies the compatibility of the antibodies to the antigens. Differences imply potential failure by the vaccine strains to recognize antigens upon infection.

**VP7 Epitopes**
<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/VP7_Rotarix_vaccine_epitope.png" align="center">


**VP4 Epitopes**
<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/VP4_Rotarix_epitope.png" align="center">
    
No unique epitopes observed in the vaccine strains. No changes in the epitopes after vaccine introduction.
       
#### iii. Cytotoxic VP7 T-Cell Epitopes
CD8-Tcells play an important role in clearing of RVA infection and providing protection against re-infections.
Despite the fact that the antibodies, particularly IgA, are generally regarded as the most important protective factors against rotavirus infection, the role of cell-mediated immunity, namely CD8? T lymphocytes, has also been demonstrated in several studies. Animal models are quite popular for studying rotavirus infection. In particular, a knockout mouse model with defective specific immunedependent genes resulting in a decrease in certain lymphocyte populations showed that CD8? T cells can mediate the clearance of rotavirus infection from the organism, resulting in complete short-term and partial long-term protection against reinfection. Two linear cytotoxic T lymphocyte (CTL) epitopes have previously been shown to be present on VP7 at amino acid positions 16-28 and 40-52, [Morozova et al,2015](https://doi.org/10.1007/s00705-015-2439-6)

There is growing evidence that RVA strains carrying different subgenotypes or sublineages within the same genotype may have different antigenic properties, thus potentially allowing the virus to evade the immune response. In other words, the difference between the amino acid composition of antigenic epitopes in the vaccine and the wild-type strains that circulate in the area where it is planned to introduce the vaccine may undermine its effectiveness. Since both vaccines are based on strains that were isolated in the early 1980s, it is likely that Russian RVA field strains carry alleles of genes that are different from those of the vaccine. Changes in an allele within a genotype or the occurrence of amino acid substitutions at antigenic sites can be followed up by sequencing and analyzing the amino acid sequences, [Morozova et al,2015](https://doi.org/10.1007/s00705-015-2439-6)


Analysis of Tcell epitopes only found two major differences between the vaccine and circulating strains. Figure 3.
        ![image](https://github.com/mikemwanga/MSC-Project/blob/main/Images/T_Cell%20epitopes.png)

Changes in position V42M and A46V are only observed in post-vaccine strains. Needs further analysis to determine whether this affect viral clearence. Does this affect the structure of the proteins?




### Objective 3. Determine the origin and spread patterns of the local strains
**Bayesian analysis of local data**
## Aimed at determining the Evolutionary rate and tmrca of the VP7 and VP4 genes


Tempest showed sufficient signal for the local data


<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/tempest_sufficient_signal.png" align="center">
=======
<image src="https://github.com/mikemwanga/MSC-Project/blob/main/tempest_sufficient_signal.png" align="center">
>>>>>>> dc7d6e919279794bef961b38934a0739d8531409

Performed coalescence analysis usin Beauti and Beast using the following parameters
 - Lognormal relaxed (uncorrelated) clock model (67), 
 - Constant-size coalescent tree prior, 
 - Hasegawa- Kishino-Yano (HKY85) nucleotide substitution model with estimated base frequencies (68), 
 - Site heterogeneity model with 4 rate categories 
 - Did 20 million chains

Loaded the resulting log file to Tracer. ESS values not significant

<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/tracer_output.png" align="center">







<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/Screenshot%202021-02-05%20at%2013.53.33.png" align="center">


Re-analysed the data by adding global sequences and performed 2 million MCMC chains. 
Parameters:
 - Model HKY with estimated base frequencies
 - lognormal, Uncorrelated relaxed clock model
 - Tree prior  - coalescent constant with random starting tree
 - Uniform priors for tmrca, ucld-mean and frequencies


TempEst had a strong signal

<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/tempest_run4.png" align="center">


The ESS converged satisfactorily

<image src="https://github.com/mikemwanga/MSC-Project/blob/main/Images/ESS_run4.png" align="center">

















## Change in diversity before and after VI
## Map vaccination status of your geographical location
## Map areas where G1P8 was identified
## Perfome network analysis or something
## Protein structure based on the B/A vacine changes
## Determine severity of the disease in vaccinated vs unvaccinated and in those rearssortant viruses


